### PR TITLE
feat(logs): remove logger abstraction

### PR DESCRIPTION
**Pull request by mwangggg**
fixes: #52

### CODE REVIEW

**Comment by mwangggg** [comment]
oh I didn't notice https://github.com/cryostatio/cryostat-core/pull/322 is there any overlap?

**Comment by andrewazores** [comment]
I think #52 probably also solves or is a replacement for #39, so in that case this PR also replaces #322.

**Comment by andrewazores** [comment]
Test failure looks like just a sporadic flakey test.

**Comment by andrewazores** [comment]
https://github.com/cryostatio/cryostat-core/actions/runs/7877293531

**Comment by andrewazores** [comment]
Different sporadic test failures, fun...

**Comment by mwangggg** [comment]
:o so many failures

**Comment by andrewazores** [comment]
Yea, this is getting ridiculous. Hitting every sporadic failure I've seen for the 2.x codebase in a row.

**Comment by andrewazores** [comment]
https://github.com/cryostatio/cryostat-core/actions/runs/7877597207

**Review by andrewazores — APPROVED** [review]

File: src/test/java/io/cryostat/core/net/discovery/JvmDiscoveryClientTest.java [hunk 1]
```diff
@@ -42,13 +41,16 @@
...
+import org.slf4j.LoggerFactory;
 
 @ExtendWith(MockitoExtension.class)
 class JvmDiscoveryClientTest {
 
     JvmDiscoveryClient client;
     @Mock JDPClient jdp;
-    @Mock Logger logger;
+    @Mock Logger logger = LoggerFactory.getLogger(getClass());
+    ;
```
Comment (andrewazores) [inline src/test/java/io/cryostat/core/net/discovery/JvmDiscoveryClientTest.java:54]: extra `;`

File: src/main/java/io/cryostat/core/net/discovery/JvmDiscoveryClient.java [hunk 2]
```diff
@@ -26,11 +26,12 @@
...
 import org.openjdk.jmc.jdp.client.JDPClient;
 
-import io.cryostat.core.log.Logger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class JvmDiscoveryClient {
 
-    private final Logger logger;
+    private Logger logger = LoggerFactory.getLogger(getClass());
```
Comment (andrewazores) [inline src/main/java/io/cryostat/core/net/discovery/JvmDiscoveryClient.java:34]: Could these be kept `final`?

**Review by andrewazores — CHANGES_REQUESTED** [review]
Actually, just a couple of minor nitpicks before merge.

File: src/main/java/io/cryostat/core/net/discovery/JvmDiscoveryClient.java (hunk 2 above)
Comment (mwangggg) [reply src/main/java/io/cryostat/core/net/discovery/JvmDiscoveryClient.java:34]: hmm I believe when I kept them as finals an error was produced but I can double check that now

**Review by andrewazores — APPROVED** [review]

**Pull request merged**